rclpy/test/test_parameter_client.py

```diff
@@ -0,0 +1,136 @@
+# Copyright 2022 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from tempfile import NamedTemporaryFile
+import unittest
+
+import rcl_interfaces.msg
+import rcl_interfaces.srv
+import rclpy
+import rclpy.context
+from rclpy.executors import SingleThreadedExecutor
+from rclpy.parameter import Parameter
+from rclpy.parameter_client import AsyncParameterClient
+
+
+class TestParameterClient(unittest.TestCase):
+
+    def setUp(self):
+        self.context = rclpy.context.Context()
+        rclpy.init(context=self.context)
+        self.client_node = rclpy.create_node('test_parameter_client',
+                                             namespace='/rclpy',
+                                             context=self.context)
+        self.target_node = rclpy.create_node('test_parameter_client_target',
+                                             namespace='rclpy',
+                                             allow_undeclared_parameters=True,
+                                             context=self.context)
+        self.target_node.declare_parameter('int_arr_param', [1, 2, 3])
+        self.target_node.declare_parameter('float/param//', 3.14)
+
+        self.client = AsyncParameterClient(self.client_node, 'test_parameter_client_target')
+        self.executor = SingleThreadedExecutor(context=self.context)
+        self.executor.add_node(self.client_node)
+        self.executor.add_node(self.target_node)
+
+    def tearDown(self):
+        self.executor.shutdown()
+        self.client_node.destroy_node()
+        self.target_node.destroy_node()
+        rclpy.shutdown(context=self.context)
+
+    def test_set_parameter(self):
+        future = self.client.set_parameters([
+            Parameter('int_param', Parameter.Type.INTEGER, 88).to_parameter_msg(),
+            Parameter('string/param', Parameter.Type.STRING, 'hello world').to_parameter_msg(),
+        ])
+        self.executor.spin_until_future_complete(future)
+        results = future.result()
+        assert results is not None
+        res = [i.successful for i in results.results]
+        assert all(res)
+
+    def test_get_parameter(self):
+        future = self.client.get_parameters(['int_arr_param', 'float/param//'])
+        self.executor.spin_until_future_complete(future)
+        results = future.result()
+        assert results is not None
+        assert list(results.values[0].integer_array_value) == [1, 2, 3]
```

Comment:
Not sure if there's an order guarantee? Would `float/param//` ever be the first element in the array?